teaching-platform/db_handler.py

```diff
@@ -0,0 +1,100 @@
+import logging
+
+import yaml
+import jsonpickle
+from password_strength import PasswordPolicy
+from werkzeug.security import generate_password_hash, check_password_hash
+
+import exceptions
+from models import User, CensuredUser, Role
+from extensions import db
+
+
+with open("../config.yaml", "r") as f:
+    config = yaml.load(f, Loader=yaml.FullLoader)
+
+policy = PasswordPolicy.from_names(length=5,uppercase=1, numbers=2,special=1, nonletters=2)
+
+logging.basicConfig(level=logging.INFO,filemode="w", filename="../logs.log")
+jsonpickle.set_preferred_backend('json')
+jsonpickle.set_encoder_options('json', ensure_ascii=False)
+
+def add_user(username, password):
+    test = policy.test(password)
+
+    if test: raise exceptions.PasswordToWeakError(f"Password breaks the following rules {test}.")
+    if not username or " " in username or len(username) > 89: raise exceptions.IncorrectUsername()
+
+    encoded_password = generate_password_hash(password, method="sha256")
+
+    user = User(username=username, password=encoded_password, role=Role.STUDENT.value)
+
+    results = User.query.filter_by(username=username).all()
+    if results: raise exceptions.UsernameTakenError()
+
+    db.session.add(user)
+    db.session.commit()
+
+    logging.info(f"User {username} added")
+
+def remove_user(id_to_delete):
+    results = User.query.filter_by(id=id_to_delete).first()
+
+    if not results: raise exceptions.UserDosentExistError()
+
+    User.query.filter_by(id=id_to_delete).delete()
+    db.session.commit()
+
+    logging.info(f"User with id {id_to_delete} has been removed sucessfuly.")
+
+def see_all_users():
+    results = User.query.all()
+    users = []
+    for user in results: users.append(CensuredUser(user.id, user.username, user.role))
```

Comment:
As mentioned in another comment - you can use [load_only](https://docs.sqlalchemy.org/en/14/orm/loading_columns.html#sqlalchemy.orm.load_only) or [with_entities](https://docs.sqlalchemy.org/en/14/orm/query.html#sqlalchemy.orm.Query.with_entities). 
Alternatively - you can consider adding [views](https://github.com/sqlalchemy/sqlalchemy/wiki/Views)

Reply:
I tried using `load_only` but after i get the results form db, i can't properly jsonify them. I tried using json, jsonpickle and it didn't work. I just made my own `get_json()` method in `User` and deleted `CensuredUser` instead. Using `load_only` over just fetching the whole user didn't give me any benefits so i didn't use it.